docs/data/routes/docs/nextjs/creating-components/new-component/en.md

```diff
@@ -0,0 +1,221 @@
+   
+       <RichText className="contentDescription" field={fields.BodyText} />
+     </>
+   );
+   
+   export default DataSourceExample;
+    
+   ```
+
+### Fill in values in the Experience Editor
+
+To fill in the values:
+
+1. In the Experience Editor, open `/sitecore/content/MyProject/home`.
+2. Fill in values for all fields.
+3. Click **Save** and **Publish**.
+4. Refresh `https://myproject.localhost` and see the changes.
+
+> Be aware of the following when you use the solution:
+>
+> * A publishing of the `Platform` project updates the running `cm` service.
+> * The running rendering service uses `dotnet watch` and re-compiles automatically for any changes you make. You can also run the `rendering` project directly from Visual Studio.
+
+## Using the code-first development workflow
+
+This walkthrough demonstrates how to create a new component using the code-first development workflow for your project.
+
+> This walkthrough assumes you have set up your project using the [`dotnet new`](/docs/nextjs/getting-started/walkthrough-dotnetnew) template with a project called `MyProject` or [`jss create`](/docs/nextjs/getting-started/walkthrough-jsscreate).
+>
+> You need to be aware of the following paths: 
+>
+> * the root of the **project**: 
+>   * `dotnet new` : `<path to your solutions directory>\MyProject\`.
+>   * `jss create`: `<path to your projects directory>/my-first-jss-app/`.
+> * the root of the **Next.js-based application**: 
+>   * `dotnet new` : `<path to your solutions directory>\MyProject\src\rendering\`.
+>   * `jss create`: the same as the root of the project.
+
+To create a component in a Next.js-based JSS application:
+
+1. In a terminal, in the root directory of your project run: 
+
+   * For a `dotnet new` project: 
+
+     * In the root of the project: `docker-compose up -d`.
+     * In the Next.js application root: `jss start`.
+
+   * For either type of project, in the root of the Next.js-based application, run:
+
+     ```
+     jss start
+     ```
+
+   > You can create new components while the application is running. The application will refresh when you make changes.
+   >
+   > Regardless of how you set up the project, to see your live code changes, open: `http://localhost:3000/`.
+
+2. Open a new terminal tab/window. In your application's root directory, run: 
+
+   ```
+   jss scaffold MyComponent
+   ```
+
+   > The scaffolding system will emit a simple working component. You can modify the scaffolding conventions in `src/scripts/scaffold-component.js`.
+   >
+   > The newly created component is inactive until added to a layout on a route. See step 5.
+
+3. In the application's root directory, in `sitecore/definitions/components/MyComponent.sitecore.ts`, define the component's data. For example: 
+
+   ```typescript
+   import { CommonFieldTypes, SitecoreIcon, Manifest } from '@sitecore-jss/sitecore-jss-manifest';
+   
+   export default function (manifest: Manifest): void {
```

Comment:
Hmm we should probably match the examples at least in template structure. I'll add to PR.